File: solutions/javascript/resistor-color-duo/3/resistor-color-duo.js
export function decodedValue([first, second]) {
  return colorCode(first) * 10 + colorCode(second)
}

/**
 * This is your code from the first resistor color exercise.
 */
const COLORS = ["black","brown","red","orange","yellow","green","blue","violet","grey","white"]
const colorCode = (color) => COLORS.indexOf(color)

File: solutions/ruby/resistor-color-duo/2/resistor_color_duo.rb
module ResistorColorDuo
  COLORS = %w(black brown red orange yellow green blue violet grey white)
      .each_with_index
      .to_h
      .freeze

  module_function

  def value(colors)
    colors
      .reverse
      .each_with_index
      .reduce(0) do |result, (color, index)|
        result + COLORS[color] * 10 ** index
      end

    ##
    # @example Using intermediary string
    #
    # colors.map(&COLORS.method(:fetch)).join.to_I
  end
end
